include/matx/operators/any.h

```diff
@@ -71,8 +73,17 @@ namespace detail {
       };
 
       template <typename Out, typename Executor>
-      void Exec(Out &&out, Executor &&ex) const {
-        any_impl(cuda::std::get<0>(out), a_, ex);
+      void Exec(Out &&out, Executor) const {
+        auto output_tensor = cuda::std::get<0>(out);
+        using out_tensor_t = decltype(output_tensor);
+        auto inp_ptr = thrust::device_pointer_cast(a_.Data());
+        auto result_ptr = output_tensor.Data();
+        auto op = detail::reduceOpAny<typename out_tensor_t::value_type>();
+        auto result = thrust::reduce(inp_ptr, 
+          inp_ptr + a_.TotalSize(), 
+          op.Init(), 
+          op);
+        *result_ptr = result;
```

Comment:
So I'm not really that faimliar with MatX. on line 79 I am doing a `device_pointer_cast` - IIUC the `Executor` is what determines if it's on host or device. I changed `Executor ex` to `Executor` to get past the warning as error that it wasn't used.

Reply:
Hi @ZelboK, the `Exec` functions take any generic operator and calls the transform function. In this case there's a separate transform overload for a CUDA executor and a host executor. Since the input can be any type of operator and not just a tensor, there may not be a `Data()` method since it doesn't have to have memory backing it. For example, a user could do:

```
(a = any(ones<int>({4,4}))).run();
```

`ones` has no `Data` function since it has no memory backing it. So for this to work properly it would have to use thrust's iterator interface and wrap the operator in that as we do in other transforms. This is not a trivial change and can be a bit difficult if you're not familiar with the library.

Reply:
@cliffburdick I've contributed a bit to CCCL actually so I think I should be fine in that regard. That makes sense to me so far, I'll have to try and actually implement it to see how it goes. I presume you are referring to `make_reduce_iterator` for example. Curious to know why this isn't trivial though, is there something I'm missing? 

Tangentially, is there a way to severely reduce compile times? The feedback loop right now takes quite a long time. 

I run with these options 

```
cmake -DMATX_BUILD_TESTS=ON -DMATX_BUILD_BENCHMARKS=OFF -DMATX_BUILD_EXAMPLES=OFF -DMATX_BUILD_DOCS=OFF ..
```
and have tried commenting out tests but it still takes a long time before I actually get to see errors from the compiler.

Reply:
Hi @ZelboK, it might not be too hard then if you're familiar with CCCL and their iterators. We have our own iterator classes in iterator.h. Typically the way we use them is like this:

https://github.com/NVIDIA/MatX/blob/main/include/matx/transforms/cub.h#L720

We write a lambda to perform the function (CUB in this case), and `ReduceInput` wraps it in iterators and collapses it. The tricky part might be that we have not done this with thrust though, so I don't know if our iterators are missing something to get it to work there. 

To reduce compile times you shouldn't build everything each time. What I do is I take an existing example like fft_conv.cu (or make a new one), put my code in there that I'm testing, and compile just that with something like `make fft_conv`. That should compile in just about 10 seconds on most machines, whereas compiling everything can approach an hour on weaker machines.